Block Comments: use TextareaAutosize for comment field

adamsilverstein opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
Makes the comment input resize to the contained text.

Closes:  https://github.com/WordPress/gutenberg/issues/71425

<!-- In a few words, what is the PR actually doing? -->

## Why?
Currently the comment box has a fixed size. Instead, the textarea should scale to the contained comment text with a minimum size similar to the current display.


## How?
Use the TextareaAutosize component.


## Testing Instructions
1. make sure inline commenting experiment is enabled
2. leave a comment that is much larger than the input box

expected results: 
The text area should expand to show the full text. 

### Testing Instructions for Keyboard
same

hbhalodia committed:
Change to text area control to add the new comments

hbhalodia committed:
Change to text area control to add the new comments

adamsilverstein committed:
try autosize text  area

Mamaduka commented on packages/editor/src/components/collab-sidebar/comment-form.js:
```diff
@@ -31,13 +32,12 @@ function CommentForm( { onSubmit, onCancel, thread, submitButtonText } ) {
+				onChange={ ( comment ) =>
+					setInputComment( comment.target.value )
+				}
 				label={ __( 'Comment' ) }
```
A couple of notes:

* The `TextareaAutosize` doesn't handle rendering labels for us. Let's handle it ourselves - `<VisuallyHidden as="label" htmlFor={ inputId }>`.
* I think we can use `rows={ 4 }` to alter minimum visible. There's also `maxRows`, which allows us to set an upper limit.

adamsilverstein replied:
- added rows= { 4 }
- set maxRows to 20
- added label using VisuallyHidden component (does that look right?)

Mamaduka replied:
That looks right, thank you, @adamsilverstein!

We should also remove the leftover `label` prop from `TextareaAutosize` since it does nothing.

adamsilverstein replied:
> We should also remove the leftover label prop from TextareaAutosize since it does nothing 

right!

adamsilverstein replied:
done!

Mamaduka commented:
I wish `field-sizing` had better browser support so that we could drop JS hacks.

adamsilverstein committed:
Address feedback
- added rows= { 4 }
- set maxRows to 20
- added label using VisuallyHidden component

adamsilverstein committed:
Remove unused label prop

Mamaduka commented on packages/editor/src/components/collab-sidebar/comment-form.js:
```diff
@@ -29,16 +35,23 @@ function CommentForm( { onSubmit, onCancel, thread, submitButtonText } ) {
+				<VisuallyHidden as="label" htmlFor={ inputId }>
+					{ __( 'Comment' ) }
+				</VisuallyHidden>
+			</TextareaAutosize>
```
Sorry, I just noticed this. I thought it was common practice to render labels next to a `textarea` and not as its content. Even with `input`, you can wrap it in the `label` and not the other way around.

Example: https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/textarea

adamsilverstein replied:
Ah, good point. I will adjust.

adamsilverstein replied:
I tried wrapping with the label, but the input wound up being hidden:

<img width="570" height="310" alt="image" src="https://github.com/user-attachments/assets/baf18972-4c2c-427a-95a1-d74ec5ede64e" />

Putting the label before the input worked fine though, I posted that for review.

adamsilverstein replied:
posted in https://github.com/WordPress/gutenberg/pull/71424/commits/e30ebf38955e2f57d017ad08ab2417595b269f03 along with a bunch of other commits after a rebase inadvertently.

Mamaduka replied:
There’s no to wrap anything. We can put components side by side, the ID will be used for connection.

adamsilverstein committed:
try label before input

Mamaduka reviewed (approved):
Thanks, @adamsilverstein!

Did you figure out what the reason was?
